onnxruntime/core/providers/openvino/backend_utils.cc

```diff
@@ -39,21 +39,21 @@ struct static_cast_int64 {
   int64_t operator()(const T1& x) const { return static_cast<int64_t>(x); }
 };
 
-std::shared_ptr<OVNetwork>
+std::shared_ptr<const OVNetwork>
 CreateOVModel(const ONNX_NAMESPACE::ModelProto& model_proto, const GlobalContext& global_context,
               std::map<std::string, std::shared_ptr<ov::Node>>& const_outputs_map) {
   if (IsCILogEnabled()) {
     std::cout << "CreateNgraphFunc" << std::endl;
   }
   const std::string model = model_proto.SerializeAsString();
   try {
-    auto cnn_network = global_context.ie_core.ReadModel(model, global_context.onnx_model_path_name);
+    auto ov_model = global_context.ie_core.ReadModel(model, global_context.onnx_model_path_name);
 
     // Check for Constant Folding
-    if (!global_context.is_wholly_supported_graph) {
+    if ((global_context.device_type != "NPU") && !global_context.is_wholly_supported_graph) {
```

Comment:
(global_context.device_type != "NPU") fragments the implementation per device. 

@sfatimar, @preetha-intel..  Have checked that eliminating constant folding pass does not impact NPU, but need to verify its applicability else needs to be optimized out.

Reply:
We do not support any subgraph partitions in production

Reply:
@MayureshV1 Thanks for clarifying that